src/profile-logic/convert-markers.js

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+export function upgradeGCMinorMarker(marker: Object) {
+  if ('nursery' in marker) {
+    if ('status' in marker.nursery) {
+      if (marker.nursery.status === 'no collection') {
+        marker.nursery.status = 'nursery empty';
+      }
+    } else {
+      /*
+       * This is the old format for GCMinor, rename some
+       * properties to the more sensible names in the newer
+       * format and set the status.
+       *
+       * Note that we don't delete certain properties such as
+       * promotion_rate, leave them so that anyone opening the
+       * raw json data can still see them in converted profiles.
+       */
+      marker.nursery.status = 'complete';
+
+      marker.nursery.bytes_used = marker.nursery.nursery_bytes;
+      delete marker.nursery.nursery_bytes;
+
+      // cur_capacity cannot be filled in.
+      marker.nursery.new_capacity = marker.nursery.new_nursery_bytes;
+      delete marker.nursery.new_nursery_bytes;
+
+      marker.nursery.phase_times = marker.nursery.timings;
+      delete marker.nursery.timings;
+    }
+  }
+}
+
+/*
+ * Fix the units for GCMajor and GCSlice phase times.
+ */
+export function convertPhaseTimes(old_phases: Object): Object {
+  const phases = {};
+  for (const phase in old_phases) {
+    phases[phase] = old_phases[phase] * 1000;
+  }
+  return phases;
+}
+
+/*
+ * Upgrade a GCMajor marker.  Set processed_profile to true for a processed
+ * profile, or false for a gecko profile.
+ */
+export function upgradeGCMajorMarker(
```

Comment:
I think this function should be split into two functions: `upgradeGCMajorMarker_Gecko8To9` and `upgradeGCMajorMarker_Gecko9ToProcessed9`. Then you can use two different flow types for their return values (e.g. `GCMajorMarkerPayloadGecko` and `GCMajorMarkerPayload`), and you can call `upgradeGCMajorMarker_Gecko9ToProcessed9` from `_processMarkers`.

I think having two different types for the marker payload format, i.e. one for the Gecko format and one for the processed format, is important. You'll want to have a place to put a comment about the fact that in the Gecko form, `mmu_20ms` is a percentage value, and in the processed form it's a fraction.